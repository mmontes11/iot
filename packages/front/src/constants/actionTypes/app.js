export const TOGGLE_HAMBURGER_MENU = "TOGGLE_HAMBURGER_MENU";
export const TOGGLE_MAP_DIALOG = "TOGGLE_MAP_DIALOG";
export const SET_SHOW_ERROR = "SET_SHOW_ERROR";
