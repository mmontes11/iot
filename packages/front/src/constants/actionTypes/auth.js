export const IS_AUTH = "IS_AUTH";
export const SET_USERNAME = "SET_USERNAME";
export const SET_PASSWORD = "SET_PASSWORD";
export const LOGIN_REQUEST = "LOGIN_REQUEST";
export const LOGIN_REQUEST_SUCCESS = "LOGIN_REQUEST_SUCCESS";
export const LOGIN_REQUEST_ERROR = "LOGIN_REQUEST_ERROR";
