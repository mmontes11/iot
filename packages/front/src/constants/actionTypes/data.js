export const DATA_UPDATED = "DATA_UPDATED";
export const DATA_REQUEST = "DATA_REQUEST";
export const DATA_REQUEST_SUCCESS = "DATA_REQUEST_SUCCESS";
export const DATA_REQUEST_ERROR = "DATA_REQUEST_ERROR";
export const ADD_DATA_ITEM = "ADD_DATA_ITEM";
export const RESET_DATA = "RESET_DATA";
