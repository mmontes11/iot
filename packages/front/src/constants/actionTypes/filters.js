export const FILTER_TYPE_SELECT = "FILTER_TYPE_SELECT";
export const ADD_FILTER_TYPE = "ADD_FILTER_TYPE";
export const DELETE_FILTER_TYPE = "DELETE_FILTER_TYPE";
