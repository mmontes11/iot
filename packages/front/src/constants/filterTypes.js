export const THING_FILTER_TYPE = "thing";
export const DATE_FILTER_TYPE = "date";
export const FILTER_TYPES = [THING_FILTER_TYPE, DATE_FILTER_TYPE];
