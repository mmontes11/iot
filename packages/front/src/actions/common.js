import { RESET } from "constants/actionTypes/common";

export const reset = preserveError => dispatch => {
  dispatch({ type: RESET, preserveError });
};
