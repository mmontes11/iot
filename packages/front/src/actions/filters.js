import { FILTER_TYPE_SELECT, ADD_FILTER_TYPE, DELETE_FILTER_TYPE } from "constants/actionTypes/filters";

export const selectFilterType = () => dispatch => {
  dispatch({ type: FILTER_TYPE_SELECT });
};

export const addFilterType = addedFilterType => dispatch => {
  dispatch({ type: ADD_FILTER_TYPE, addedFilterType });
};

export const deleteFilterType = deletedFilterType => dispatch => {
  dispatch({ type: DELETE_FILTER_TYPE, deletedFilterType });
};
