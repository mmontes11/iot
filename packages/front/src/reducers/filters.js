import { FILTER_TYPE_SELECT, ADD_FILTER_TYPE, DELETE_FILTER_TYPE } from "constants/actionTypes/filters";
import { RESET } from "constants/actionTypes/common";
import { THING_FILTER_TYPE, DATE_FILTER_TYPE, FILTER_TYPES } from "constants/filterTypes";
import thingFilterReducer, { initialState as thingFilterInitialState } from "reducers/thingFilter";
import dateFilterReducer, { initialState as dateFilterInitialState } from "reducers/dateFilter";

export const initialState = {
  type: {
    items: FILTER_TYPES,
    isActive: false,
    isDisabled: false,
  },
  thingFilter: thingFilterInitialState,
  dateFilter: dateFilterInitialState,
  items: [],
};

export default (state = initialState, action) => {
  switch (action.type) {
    case FILTER_TYPE_SELECT:
      return { ...state, type: { ...state.type, isActive: !state.type.isActive } };
    case ADD_FILTER_TYPE: {
      const { addedFilterType } = action;
      const addedItems = [...state.items];
      if (!addedItems.includes(addedFilterType)) {
        addedItems.push(addedFilterType);
      }
      const filterTypeItems = state.type.items.filter(item => !addedItems.includes(item));
      const allItemsSelected = FILTER_TYPES.every(item => addedItems.includes(item));
      return {
        ...state,
        type: {
          ...state.type,
          items: filterTypeItems,
          isActive: false,
          isDisabled: allItemsSelected,
        },
        items: addedItems,
      };
    }
    case DELETE_FILTER_TYPE: {
      const { deletedFilterType } = action;
      const addedItems = [...state.items];
      const newItems = addedItems.filter(item => item !== deletedFilterType);
      const filterTypeItems = [...state.type.items];
      if (!filterTypeItems.includes(deletedFilterType)) {
        filterTypeItems.push(deletedFilterType);
      }
      const allItemsSelected = FILTER_TYPES.every(item => newItems.includes(item));
      const nextState = {
        ...state,
        type: {
          ...state.type,
          items: filterTypeItems,
          isDisabled: allItemsSelected,
        },
        items: newItems,
      };
      if (deletedFilterType === THING_FILTER_TYPE) {
        return {
          ...nextState,
          thingFilter: thingFilterInitialState,
        };
      } else if (deletedFilterType === DATE_FILTER_TYPE) {
        return {
          ...nextState,
          dateFilter: dateFilterInitialState,
        };
      }
      return nextState;
    }
    case RESET:
      return initialState;
    default:
      return {
        ...state,
        thingFilter: thingFilterReducer(state.thingFilter, action),
        dateFilter: dateFilterReducer(state.dateFilter, action),
      };
  }
};
