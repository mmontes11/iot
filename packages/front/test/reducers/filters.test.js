import deepFreeze from "deep-freeze";
import filtersReducer from "reducers/filters";
import { RESET } from "constants/actionTypes/common";
import { FILTER_TYPE_SELECT, ADD_FILTER_TYPE, DELETE_FILTER_TYPE } from "constants/actionTypes/filters";
import { THING_FILTER_TYPE, DATE_FILTER_TYPE, FILTER_TYPES } from "constants/filterTypes";
import { initialState } from "../constants/index";

const filtersInitialState = initialState.filters;
deepFreeze(filtersInitialState);

describe("reducers/filters", () => {
  it("has a default state", () => {
    expect(filtersReducer(undefined, { type: "@init" })).toMatchSnapshot();
  });
  it("reduces nothing", () => {
    expect(filtersReducer(filtersInitialState, { type: "WHATEVER" })).toMatchSnapshot();
  });
  it("reduces FILTER_TYPE_SELECT", () => {
    expect(filtersReducer(filtersInitialState, { type: FILTER_TYPE_SELECT })).toMatchSnapshot();
  });
  it("reduces ADD_FILTER_TYPE", () => {
    const state = {
      ...filtersInitialState,
      items: [THING_FILTER_TYPE],
    };
    deepFreeze(state);
    expect(filtersReducer(state, { type: ADD_FILTER_TYPE, addedFilterType: THING_FILTER_TYPE })).toMatchSnapshot();
    expect(filtersReducer(state, { type: ADD_FILTER_TYPE, addedFilterType: DATE_FILTER_TYPE })).toMatchSnapshot();
  });
  it("reduces DELETE_FILTER_TYPE", () => {
    const state = {
      ...filtersInitialState,
      items: FILTER_TYPES,
    };
    deepFreeze(state);
    expect(filtersReducer(state, { type: DELETE_FILTER_TYPE, deletedFilterType: THING_FILTER_TYPE })).toMatchSnapshot();
    expect(filtersReducer(state, { type: DELETE_FILTER_TYPE, deletedFilterType: DATE_FILTER_TYPE })).toMatchSnapshot();
    expect(filtersReducer(state, { type: DELETE_FILTER_TYPE, deletedFilterType: "WHATEVER" })).toMatchSnapshot();
  });
  it("reduces RESET", () => {
    expect(filtersReducer(filtersInitialState, { type: RESET })).toMatchSnapshot();
  });
});
