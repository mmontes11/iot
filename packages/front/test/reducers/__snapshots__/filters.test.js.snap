// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/filters has a default state 1`] = `
Object {
  "dateFilter": Object {
    "custom": Object {
      "endDate": null,
      "startDate": null,
    },
    "isCustomSelected": false,
    "timePeriod": Object {
      "isActive": false,
      "isDisabled": false,
      "isLoading": false,
      "items": Array [],
      "selectedItem": null,
    },
  },
  "items": Array [],
  "thingFilter": Object {
    "isActive": false,
    "isDisabled": false,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "type": Object {
    "isActive": false,
    "isDisabled": false,
    "items": Array [
      "thing",
      "date",
    ],
  },
}
`;

exports[`reducers/filters reduces ADD_FILTER_TYPE 1`] = `
Object {
  "dateFilter": Object {
    "custom": Object {
      "endDate": null,
      "startDate": null,
    },
    "isCustomSelected": false,
    "timePeriod": Object {
      "isActive": false,
      "isDisabled": false,
      "isLoading": false,
      "items": Array [],
      "selectedItem": null,
    },
  },
  "items": Array [
    "thing",
  ],
  "thingFilter": Object {
    "isActive": false,
    "isDisabled": false,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "type": Object {
    "isActive": false,
    "isDisabled": false,
    "items": Array [
      "date",
    ],
  },
}
`;

exports[`reducers/filters reduces ADD_FILTER_TYPE 2`] = `
Object {
  "dateFilter": Object {
    "custom": Object {
      "endDate": null,
      "startDate": null,
    },
    "isCustomSelected": false,
    "timePeriod": Object {
      "isActive": false,
      "isDisabled": false,
      "isLoading": false,
      "items": Array [],
      "selectedItem": null,
    },
  },
  "items": Array [
    "thing",
    "date",
  ],
  "thingFilter": Object {
    "isActive": false,
    "isDisabled": false,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "type": Object {
    "isActive": false,
    "isDisabled": true,
    "items": Array [],
  },
}
`;

exports[`reducers/filters reduces DELETE_FILTER_TYPE 1`] = `
Object {
  "dateFilter": Object {
    "custom": Object {
      "endDate": null,
      "startDate": null,
    },
    "isCustomSelected": false,
    "timePeriod": Object {
      "isActive": false,
      "isDisabled": false,
      "isLoading": false,
      "items": Array [],
      "selectedItem": null,
    },
  },
  "items": Array [
    "date",
  ],
  "thingFilter": Object {
    "isActive": false,
    "isDisabled": false,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "type": Object {
    "isActive": false,
    "isDisabled": false,
    "items": Array [
      "thing",
      "date",
    ],
  },
}
`;

exports[`reducers/filters reduces DELETE_FILTER_TYPE 2`] = `
Object {
  "dateFilter": Object {
    "custom": Object {
      "endDate": null,
      "startDate": null,
    },
    "isCustomSelected": false,
    "timePeriod": Object {
      "isActive": false,
      "isDisabled": false,
      "isLoading": false,
      "items": Array [],
      "selectedItem": null,
    },
  },
  "items": Array [
    "thing",
  ],
  "thingFilter": Object {
    "isActive": false,
    "isDisabled": false,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "type": Object {
    "isActive": false,
    "isDisabled": false,
    "items": Array [
      "thing",
      "date",
    ],
  },
}
`;

exports[`reducers/filters reduces DELETE_FILTER_TYPE 3`] = `
Object {
  "dateFilter": Object {
    "custom": Object {
      "endDate": null,
      "startDate": null,
    },
    "isCustomSelected": false,
    "timePeriod": Object {
      "isActive": false,
      "isDisabled": false,
      "isLoading": false,
      "items": Array [],
      "selectedItem": null,
    },
  },
  "items": Array [
    "thing",
    "date",
  ],
  "thingFilter": Object {
    "isActive": false,
    "isDisabled": false,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "type": Object {
    "isActive": false,
    "isDisabled": true,
    "items": Array [
      "thing",
      "date",
      "WHATEVER",
    ],
  },
}
`;

exports[`reducers/filters reduces FILTER_TYPE_SELECT 1`] = `
Object {
  "dateFilter": Object {
    "custom": Object {
      "endDate": null,
      "startDate": null,
    },
    "isCustomSelected": false,
    "timePeriod": Object {
      "isActive": false,
      "isDisabled": false,
      "isLoading": false,
      "items": Array [],
      "selectedItem": null,
    },
  },
  "items": Array [],
  "thingFilter": Object {
    "isActive": false,
    "isDisabled": false,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "type": Object {
    "isActive": true,
    "isDisabled": false,
    "items": Array [
      "thing",
      "date",
    ],
  },
}
`;

exports[`reducers/filters reduces RESET 1`] = `
Object {
  "dateFilter": Object {
    "custom": Object {
      "endDate": null,
      "startDate": null,
    },
    "isCustomSelected": false,
    "timePeriod": Object {
      "isActive": false,
      "isDisabled": false,
      "isLoading": false,
      "items": Array [],
      "selectedItem": null,
    },
  },
  "items": Array [],
  "thingFilter": Object {
    "isActive": false,
    "isDisabled": false,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "type": Object {
    "isActive": false,
    "isDisabled": false,
    "items": Array [
      "thing",
      "date",
    ],
  },
}
`;

exports[`reducers/filters reduces nothing 1`] = `
Object {
  "dateFilter": Object {
    "custom": Object {
      "endDate": null,
      "startDate": null,
    },
    "isCustomSelected": false,
    "timePeriod": Object {
      "isActive": false,
      "isDisabled": false,
      "isLoading": false,
      "items": Array [],
      "selectedItem": null,
    },
  },
  "items": Array [],
  "thingFilter": Object {
    "isActive": false,
    "isDisabled": false,
    "isLoading": false,
    "items": Array [],
    "selectedItem": null,
  },
  "type": Object {
    "isActive": false,
    "isDisabled": false,
    "items": Array [
      "thing",
      "date",
    ],
  },
}
`;
